### DISCUSSION TITLE
Need help with `c.executionCtx.passThroughOnException()`

### DISCUSSION THREAD

**Discussion by AndriusJonaitis** [honojs/hono#1178] [General]
Hey,

I've encountered an issue with `c.executionCtx.passThroughOnException()`. In my Hono setup, it doesn't prevent the `Internal Server Error` from appearing in the browser, as I believe it should.

I initialized the worker with `npm create hono@latest my-app`. The error persists even when the code is updated as follows:

```js
app.get('/', (c) => {
    c.executionCtx.passThroughOnException()

    throw new Error('!')

    return c.text('Hello Hono!')
})
```

However, when I refactored my code without Hono, `passThroughOnException` functioned as expected, preventing the `Internal Server Error`:

```js
export default {
    async fetch(request: Request, env: any, ctx: ExecutionContext) {
        ctx.passThroughOnException()

        throw new Error('!')

        const html = `<!DOCTYPE html>
            <body>
                <h1>Hello Hono!</h1>
            </body>`;
    
        return new Response(html, {
            headers: {
                "content-type": "text/html;charset=UTF-8",
            },
        });
    },
};
```

Why does `passThroughOnException` get ignored with a Hono setup?

**Comment by yusukebe** [comment]
If your purpose is simply to avoid showing the "Internal Server Error" to users, you can use `onError()`.

https://hono.dev/api/hono#error-handling

```ts
app.onError((e, c) => {
  console.log(e)
  return c.text('!', 500)
})
```

**Comment by toinbis** [comment]
Hey @yusukebe !

Thanks a lot for reply. https://developers.cloudflare.com/workers/runtime-apis/fetch-event/#passthroughonexception is crucial functionality. It allows team to develop worker being confident that whatever happens - this command prevents from any bad stuff happenning.

Can't it be replicated to behave identically as in CF workers?

**Comment by yusukebe** [comment]
I've never used `passThroughOnException()` in production, so I'm not very familiar with it. Hence, I can't attest to its necessity. However, if you wish to use `passThroughOnException()`, you could write it as follows:

```ts
app.onError((e, _c) => {
  throw e
})

export default {
  fetch: (request: Request, env: {}, context: ExecutionContext) => {
    context.passThroughOnException()
    return app.fetch(request, env, context)
  }
}
```

Does it work as expected?
